cmd/root.go

```diff
@@ -256,6 +269,32 @@ func runSignalWrapper(cmd *Command) error {
 
 	shutdownCh := make(chan error)
 
+	if cmd.prometheusNamespace != "" {
+		e, err := prometheus.NewExporter(prometheus.Options{
+			Namespace: cmd.prometheusNamespace,
+		})
+		if err != nil {
+			return err
+		}
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", e)
+		addr := fmt.Sprintf("localhost:%s", cmd.prometheusPort)
+		server := &http.Server{Addr: addr, Handler: mux}
+		go func() {
+			select {
+			case <-ctx.Done():
+				if err := server.Shutdown(context.Background()); err != nil {
```

Comment:
Is background the right thing to use here? If so, should it have a timeout so this doesn't hang the proxy?

If ctx.Done() is already closed, doesn't that mean the server is already shutdown? Would shutdownCh already have been sent a value then? Maybe this needs to be logged indiscriminately.

Reply:
Probably not. I added a context with a 1 second timeout. And yes, logging makes more sense as you suggest.